src/librustc_typeck/check/mod.rs

```diff
@@ -999,6 +981,17 @@ impl<'a, 'gcx, 'tcx> Visitor<'gcx> for GatherLocalsVisitor<'a, 'gcx, 'tcx> {
                 _: hir::BodyId, _: Span, _: ast::NodeId) { }
 }
 
+/// When `check_fn` is invoked on a generator (i.e., a body that
+/// includes yield), it returns back some information about the yield
+/// points.
+struct GeneratorTypes<'tcx> {
+    /// Type of value that is yielded.
+    yield_ty: ty::Ty<'tcx>,
```

Comment:
Is this need given that `FnCtxt` which is also returned by `check_fn` has the yield type?

Reply:
We are creating a new `FnCtxt` inside of `check_fn` and I think the yield type is stored in there, no?

Reply:
It looks like this structure was made just so you could return `yield_ty` from `check_fn`, but that is already returned in the `FnCtxt`.

Reply:
Oh dear, it looks like it is returned in the `generator_sigs` map inside of `TypeckTables` -- is that what you mean? If so, that's because I forgot to remove that map when I removed the rest of `generator_sigs`-related things, I think.

Reply:
(It should now be obtained by using the `poly_gen_sig` method on the `ClosureSubsts`)

Reply:
So do we still want this?